code/datums/components/crafting/guncrafting.dm

```diff
@@ -2,20 +2,130 @@
 
 // PARTS //
 
+/obj/item/weaponcrafting
+	icon = 'icons/obj/improvised.dmi'
+
+/obj/item/weaponcrafting/silkstring
+	name = "silkstring"
+	desc = "A long pice of silk looks like cable coil."
+	icon = 'icons/obj/improvised.dmi'
+	icon_state = "silkstring"
+
 /obj/item/weaponcrafting/receiver
 	name = "modular receiver"
 	desc = "A prototype modular receiver and trigger assembly for a firearm."
-	icon = 'icons/obj/improvised.dmi'
 	icon_state = "receiver"
 
 /obj/item/weaponcrafting/stock
 	name = "rifle stock"
 	desc = "A classic rifle stock that doubles as a grip, roughly carved out of wood."
-	icon = 'icons/obj/improvised.dmi'
 	icon_state = "riflestock"
 
-/obj/item/weaponcrafting/silkstring
-	name = "silkstring"
-	desc = "A long pice of silk looks like cable coil."
-	icon = 'icons/obj/improvised.dmi'
-	icon_state = "silkstring"
\ No newline at end of file
+/obj/item/weaponcrafting/stock/attackby(obj/item/S, mob/user, params)
+	..()
+	if(istype(S, /obj/item/pipe))
+		return
+	if(istype(S, /obj/item/weaponcrafting/receiver))
+		var/obj/item/weaponcrafting/receiver/stock/R = /obj/item/weaponcrafting/receiver/stock
```

Comment:
Just remove this line use the Path in place of R on the `new` procedure.

This also applies to most, if not all, other steps.